src/test/java/bio/terra/service/auth/iam/sam/SamIamTest.java

```diff
@@ -709,7 +740,9 @@ public void testOverwriteGroupPolicyEmails() throws InterruptedException, ApiExc
   }
 
   @Test
-  public void testOverwriteGroupPolicyEmailsThrowsWhenSamGroupApiThrows() throws ApiException {
+  void testOverwriteGroupPolicyEmailsThrowsWhenSamGroupApiThrows() throws ApiException {
+    mockConfigService();
+    mockSamGroupApi();
```

Comment:
I don't think we do this yet, but I've thought about using JUnit 5's nested test feature to group together subtests that have a common `BeforeEach` method. In this case could have a nested test for all sam group API calls and use the nested tests's `BeforeEach` to mock the config service and the group api. See https://junit.org/junit5/docs/current/user-guide/#writing-tests-nested.

Reply:
I took a closer look and I'm pretty sure this would work well but there's a lot of tests so maybe now's not the right PR to change it. I also see a few other things that could be improved, like unnecessary use of `eq()` and unnecessary `verify()` (now that strict stubs are used). There's at least one test that has no asserts at all, `testCreateProfile()`.

Reply:
I added some nested classes! Unfortunately, there were some mocks that didn't follow a nested pattern so some of the tests are not in a nested class.